crates/ruff_linter/src/rules/pylint/rules/no_self_use.rs

```diff
@@ -102,9 +102,8 @@ pub(crate) fn no_self_use(
         .posonlyargs
         .iter()
         .chain(&parameters.args)
-        .chain(&parameters.kwonlyargs)
```

Comment:
This is unrelated to the other changes in this PR, but it's invalid for `self` to be a keyword-only parameter, so I figured I'd fix it while I was here